### PR TITLE
Set on-fire in BasicEntityRebindSupport

### PR DESCRIPTION
**Pull request by aledsage**

### CODE REVIEW

File: core/src/main/java/org/apache/brooklyn/core/mgmt/rebind/BasicEntityRebindSupport.java [hunk 1]
```diff
@@ -233,4 +237,14 @@ protected void addLocations(RebindContext rebindContext, EntityMemento memento)
             }
         }
     }
+
```
Comment (bostko) [inline core/src/main/java/org/apache/brooklyn/core/mgmt/rebind/BasicEntityRebindSupport.java:240]: `@Override`

File: core/src/main/java/org/apache/brooklyn/core/mgmt/rebind/BasicEntityRebindSupport.java [hunk 2]
```diff
@@ -233,4 +237,14 @@ protected void addLocations(RebindContext rebindContext, EntityMemento memento)
             }
         }
     }
+    
+    protected void instanceRebind(AbstractBrooklynObject instance) {
+        Preconditions.checkState(instance == entity, "Expected %s and %s to match, but different objects", instance, entity);
+        Lifecycle expectedState = ServiceStateLogic.getExpectedState(entity);
+        if (expectedState == Lifecycle.STARTING || expectedState == Lifecycle.STOPPING) {
```
Comment (bostko) [inline core/src/main/java/org/apache/brooklyn/core/mgmt/rebind/BasicEntityRebindSupport.java:244]: Considering `expectedState` is persisted then this PR makes a lot of sense.

File: core/src/main/java/org/apache/brooklyn/core/mgmt/rebind/BasicEntityRebindSupport.java (hunk 1 above)
Comment (ivanayov) [reply core/src/main/java/org/apache/brooklyn/core/mgmt/rebind/BasicEntityRebindSupport.java:240]: @bostko will address it in the next commit for separating rebind during provisioning from rebind during application starting. Merging now

**Comment by ivanayov** [comment]
Thanks @aledsage . Looks good.

**Pull request merged**